### PR TITLE
promptly restart update streams

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: exec/Miner/Updates.hs [hunk 1]
```diff
@@ -72,38 +67,39 @@ getUpdateVar (UpdateMap v) k = modifyMVar v $ \m -> case HM.lookup k m of
...
 
+-- | Atomatically restarts the stream when the response status is 2** and throws
+-- and exception otherwise.
+--
 updateStream :: ChainId -> TVar Int -> RIO Env ()
 updateStream cid var = do
     e <- ask
     u <- NEL.head <$> readIORef (envUrls e) -- Do we ever use something else than the head?
-    liftIO $ withEvents (req u) (envMgr e) $ \updates -> updates
+    forever $ liftIO $ withEvents (req u) (envMgr e) $ \updates -> updates
```
Comment (fosskers) [inline exec/Miner/Updates.hs]: I see that in [in 735](https://github.com/kadena-io/chainweb-node/pull/735/files#diff-22f2620e094b259e8e42309f48831467R174) you're returning a `CloseEvent`, but this isn't checked for in `realEvent`. Is this intentional?

Comment (larskuhtz) [reply exec/Miner/Updates.hs]: yes, after `CloseEvent` the server will terminate the HTTP request and `withEvents` will close the stream and return.

Comment (fosskers) [reply exec/Miner/Updates.hs]: And that's automatic within `wai` ?

Comment (larskuhtz) [reply exec/Miner/Updates.hs]: yes, wai does the right thing here. The question is wether `withEvents` terminates the response properly. I think, I checked and the parse will just terminate and exit the stream when it sees the end of the input stream.

**Pull request merged**